docs/user-guide/KubernetesCurator.rst

```diff
@@ -0,0 +1,382 @@
+* `ReadWriteMany <https://kubernetes.io/docs/concepts/storage/persistent-volumes/#access-modes>`__ `StorageClass <https://kubernetes.io/docs/concepts/storage/storage-classes/>`__ (setup by Kubernetes cluster admin)
+
+Storage
+-------
+To run NeMo Curator, we need to setup storage to upload and store the input
+files, as well as any processed outputs.
+
+Here is an example of how to create a dynamic PV from a StorageClass setup
+by your cluster admin. Replace ``STORAGE_CLASS=<...>`` with the name of
+your StorageClass.
+
+This example requests ``150Gi`` of space. Adjust that number for your
+workloads and be aware that not all storage provisioners support volume
+resizing.
+
+.. code-block:: bash
+
+    STORAGE_CLASS=<...>
+    PVC_NAME=nemo-workspace
+
+    kubectl apply -f - <<EOF
+    apiVersion: v1
+    kind: PersistentVolumeClaim
+    metadata:
+      name: ${PVC_NAME}
+    spec:
+      accessModes:
+        - ReadWriteMany
+      storageClassName: ${STORAGE_CLASS}
+      resources:
+        requests:
+          # Requesting enough storage for a few experiments
+          storage: 150Gi
+    EOF
+
+.. note::
+    The storage class must support ``ReadWriteMany`` because multiple Pods
+    may need to access the PVC to concurrently read and write.
+
+Setup PVC Busybox Helper Pod
+############################
+
+Inspecting the PVC and copying to and from it is facilitated with a busybox container.
+Some examples below assume you have this Pod running to copy to and from the PVC. 
+
+.. code-block:: bash
+
+    PVC_NAME=nemo-workspace
+    MOUNT_PATH=/nemo-workspace
+
+    kubectl create -f - <<EOF
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: nemo-workspace-busybox
+    spec:
+      containers:
+      - name: busybox
+        image: busybox
+        command: ["sleep", "infinity"]
+        volumeMounts:
+        - name: workspace
+          mountPath: ${MOUNT_PATH}
+      volumes:
+      - name: workspace
+        persistentVolumeClaim:
+          claimName: ${PVC_NAME}
+    EOF
+
+Feel free to delete this container if no longer needed, but it should use very little
+resources when idle.
+
+.. code-block:: bash
+
+    kubectl delete pod nemo-workspace-busybox
+
+Setup Docker Secrets
+--------------------
+A Kubernetes Secret needs to be created on the k8s cluster to authenticate with the NGC
+private registry. If not done already, get an NGC key from ngc.nvidia.com. Create a
+secret key on the k8s cluster with (replace ``<NGC KEY HERE>`` with your NGC secret
+key. Note that if you have any special characters in your key you might need to wrap
+the key in single quotes (``'``) so it can be parsed correctly by k8s)::
+
+    kubectl create secret docker-registry ngc-registry --docker-server=nvcr.io --docker-username=\$oauthtoken --docker-password=<NGC KEY HERE>
+
+Setup Python Environment
+------------------------
+
+Setup a virtual environment:
```

Comment:
Could you expand on what this environment is used for? Users might get confused and wonder if the need to install NeMo Curator in the environment or just dask kubernetes.